### PR TITLE
test[CL]: getTickInfo table driven tests (5/6)

### PR DESCRIPTION
**Pull request by czarcas7ic**
Partially Closes: #3157 

## What is the purpose of the change

Adds table driven test for getTickInfo

Fulfills following requirements as laid out in the issue:
- [x] tests added
- [x] coverage maximized
- [x] godocs added
- [x] aim for 1 method per PR

### CODE REVIEW

File: x/concentrated-liquidity/tick_test.go [hunk 1]
```diff
@@ -263,3 +264,51 @@ func (s *KeeperTestSuite) TestInitOrUpdateTick() {
...
+			expectedTickInfo: model.TickInfo{LiquidityGross: DefaultLiquidityAmt, LiquidityNet: DefaultLiquidityAmt.Neg()},
+		},
+		{
+			name:             "Get tick info on existing pool with no existing tick",
+			poolToGet:        validPoolId,
+			tickToGet:        DefaultCurrTick.Int64() + 1,
+			expectedTickInfo: model.TickInfo{LiquidityGross: sdk.ZeroDec(), LiquidityNet: sdk.ZeroDec()},
+		},
+		{
+			name:             "Get tick info on a non-existing pool with no existing tick",
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick_test.go:288]: Why do we not return an error in such a case instead? Doesn't make sense to return a valid tick info with zero values in such a case

Comment (czarcas7ic) [reply x/concentrated-liquidity/tick_test.go]: Okay, I added a poolExists check. Basically if the pool does not exist, it returns an error with nil values for tickInfo. If the pool exists but the ticks just arent initialized, it returns a valid tickInfo object but with zero values for liquidity

File: x/concentrated-liquidity/tick.go [hunk 2]
```diff
@@ -57,6 +57,9 @@ func (k Keeper) getTickInfo(ctx sdk.Context, poolId uint64, tickIndex int64) (ti
 	store := ctx.KVStore(k.storeKey)
 	tickStruct := model.TickInfo{}
 	key := types.KeyTick(poolId, tickIndex)
+	if !k.poolExists(ctx, poolId) {
+		return tickInfo, types.PoolNotFoundError{PoolId: poolId}
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick.go:61]: nit: I know that these are equivalent. However, the change guarantees that we return an empty struct even if someone accidentally modifies the `tickInfo` prior to returning

```suggestion
		return model.TickInfo{}, types.PoolNotFoundError{PoolId: poolId}
```

**Review by p0mvn — APPROVED** [review]
Great work!

**Pull request merged**